### PR TITLE
MA-142 Add informal breach agreement flow

### PR DESCRIPTION
**Pull request by maysakanoni**
**What?**

- Add send_court_agreement_breach_letter classification

**Why?**

- So that the classifications can differentiate between informal and and court ordered agreement

### CODE REVIEW

File: app/models/hackney/income/models/case_priority.rb [hunk 1]
```diff
@@ -4,7 +4,8 @@ module Models
       class CasePriority < ApplicationRecord
         enum classification: {
           no_action: 0, send_letter_two: 1, send_letter_one: 2, send_first_SMS: 3, send_NOSP: 4,
-          apply_for_court_date: 6, send_court_warning_letter: 7, send_court_agreement_breach_letter: 8
+          apply_for_court_date: 6, send_court_warning_letter: 7, send_court_agreement_breach_letter: 8,
+          send_informal_agreement_breach_letter: 9
```
Comment (richrace) [inline app/models/hackney/income/models/case_priority.rb]: @georgeschena and @maysakanoni FYI: be aware of using the same Enum value.

https://github.com/LBHackney-IT/lbh-income-api/pull/293/files#diff-83dba773cecaa571d2bf60560a95e691R8

File: lib/hackney/income/tenancy_prioritiser/tenancy_classification.rb [hunk 2]
```diff
@@ -36,10 +37,22 @@ def validate_wanted_action(wanted_action)
           raise ArgumentError, "Tried to classify a case as #{wanted_action}, but this is not on the list of valid classifications."
         end
 
+        def send_informal_agreement_breach_letter?
+          return false if @criteria.number_of_broken_agreements < 1
```
Comment (richrace) [inline lib/hackney/income/tenancy_prioritiser/tenancy_classification.rb]: Can it be a negative? If not: 

```suggestion
          return false if @criteria.number_of_broken_agreements.zero?
```

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/send_informal_agreement_breach_letter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
+require 'rails_helper'
+
+describe 'Send Informal Agreement Breach Letter Rule', type: :feature do
```
Comment (richrace) [inline spec/lib/hackney/income/tenancy_prioritiser/classifications/send_informal_agreement_breach_letter_spec.rb]: See @kjdchapman's comment here: https://github.com/LBHackney-IT/lbh-income-api/pull/293#discussion_r360393688

File: app/models/hackney/income/models/case_priority.rb (hunk 1 above)
Comment (georgeschena) [reply app/models/hackney/income/models/case_priority.rb]: Yeah, on my branch its 
`  no_action: 0, send_letter_two: 1, send_letter_one: 2, send_first_SMS: 3, send_NOSP: 4,
          apply_for_court_date: 6, send_court_warning_letter: 7, update_court_outcome_action: 8,
          send_court_agreement_breach_letter: 9`

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/send_informal_agreement_breach_letter_spec.rb (hunk 3 above)
Comment (maysakanoni) [reply spec/lib/hackney/income/tenancy_prioritiser/classifications/send_informal_agreement_breach_letter_spec.rb]: Thanks, this was super useful. I've now updated my test

**Pull request merged**